.github/tdd.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+import json
+import logging
+import os
+import subprocess
+import sys
+
+ORG_NAME = "pulp"
+REPO_NAME = "pulp_ansible"
+APP_NAME = "pulp_ansible"
+
+
+def get_current_branch():
+    pid = subprocess.run(
+        "git branch --show-current", shell=True, stdout=subprocess.PIPE, check=True
+    )
+    branch_name = pid.stdout.decode("utf-8")
+    branch_name = branch_name.strip()
+    return branch_name
+
+
+def get_changed_files(pr_branch, target_branch="master"):
+    if os.environ.get("GITHUB_PR_NUMBER"):
+        url = f"https://patch-diff.githubusercontent.com/raw/{ORG_NAME}/{REPO_NAME}"
+        url += f"/pull/{os.environ['GITHUB_PR_NUMBER']}.diff"
+        print(f"GET diff from {url}")
+        cmd = f"curl -s -o /tmp/pr.diff {url}"
+        pid = subprocess.run(cmd, shell=True)
+        assert pid.returncode == 0
+        with open("/tmp/pr.diff", "r") as f:
+            raw = f.read()
+        filenames = raw.split("\n")
+        filenames = [x for x in filenames if x.startswith("---") or x.startswith("+++")]
+        filenames = [x.split(None, 1)[1] for x in filenames]
+        filenames = [x for x in filenames if not x.startswith("/dev/null")]
+        filenames = [x.lstrip("a/") for x in filenames]
+        filenames = [x.lstrip("b/") for x in filenames]
+        filenames = sorted(set(filenames))
+    else:
+        # diffs the local checkout ...
+        cmd = f'git diff --name-only "{pr_branch}".."{target_branch}"'
+        pid = subprocess.run(cmd, shell=True, stdout=subprocess.PIPE, check=True)
+        filenames = pid.stdout.decode("utf-8")
+        filenames = filenames.split("\n")
+        filenames = [x.strip() for x in filenames if x.strip()]
+    return filenames
+
+
+def verify_test_files_changed(changed_files):
+
+    # places where appplication source goes
+    app_paths = [f"{APP_NAME}/"]
```

Comment:
I think there are some files that shouldn't require test changes.

Also, what should happen in the case where a thing is deprecated in `pulpcore.plugin` and the change is simply switching over to the new one?